supergood.go

```diff
@@ -51,6 +51,10 @@ func New(o *Options) (*Service, error) {
 	}
 
 	sg.DefaultClient = sg.Wrap(http.DefaultClient)
+
+	// Overrides default client
+	http.DefaultClient = sg.DefaultClient
+
```

Comment:
Added this by default, should consider being able to toggle this

Reply:
I think this is a little too eager (though I totally understand your desire to do it!).

Most libraries and APIs in go allow you to configure the HTTP client to use, and I'd expect most mature code-bases to already override `http.DefaultClient` (or pass a specific client to each dependency) to deal with things like setting timeouts, and other logging requirements.

Unlike in Python or Ruby where idiomatically, as much as possible is pushed to be implicit, go tends to err on the side of explicitness; so as a user of this library (who's done a bunch of go) I would shake my fist at this and set DisableDefaultClient.

The other advantage the explicit client gives you is less need for configuring which requests to include or not, you just pass the supergood client or the http client.

On the flip side, I totally get that you want things to just work by default, it just feels a bit non-go.

Reply:
I definitely defer to you on this - I actually had another Go programmer mention the same awhile back. After looking through your code, I can see the rampant explicitness of Go, so this makes sense.